In [1]:
import os
import sys
sys.path.append(os.path.abspath('.'))
sys.path.append(os.path.abspath(r'./.'))

In [2]:
import torch
torch.cuda.empty_cache()

/home/borela/miniconda3/envs/exqa-complearning/lib/python3.7/site-packages/tqdm/auto.py:22: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


In [3]:
%pwd

'/home/borela/fontes/exqa-complearning/source/calculation'

In [4]:
%cd ../..

/home/borela/fontes/exqa-complearning


In [5]:
from source.calculation.context_learning  import teste_trata_reader_pipie

In [8]:
from source.calculation.context_learning import trata_reader_pipie 

In [18]:
texto_pergunta = 'Qual o melhor time do Brasil?'
texto_contexto = 'Por causa da chuva, o Flamengo, melhor time do Brasil, ficou sem jogar a final da Libertadores. Há muitos torcedores pelo Flamengo, o melhor time, na minha Família. Além da bicicleta, em 1990, eu tinha a honra de ser flamenguista.  Ontem, quando reli os documentos do tribunal, descobri que em 1990 quando tomei posse, minha declaração de bens só continha uma bicicleta.'

In [15]:

parm_dict_config = {
                # parâmetros complementares
                "num_top_k": 1,
                "num_doc_stride": 0,
                "num_factor_multiply_top_k": 0,
                "if_handle_impossible_answer":False,
                "num_max_answer_length":200,
                # parâmetros context
                    'answer_stop': ['.', '\n', '\n','!'],
                    'generator_sample': True,
                    "val_length_penalty":0,                    
                    'generator_temperature': 1,
                    'prompt_format': ('''Instrução: Usando palavras do texto abaixo, responda à pergunta:

Texto: {context}

Pergunta: {question}
Resposta:''')}

In [9]:
name_model = "EleutherAI/gpt-j-6B"
reader = trata_reader_pipie.Reader(name_model, parm_dict_config)
print(f"reader.info()\n {reader.info}")

reader.info()
 {'name': 'EleutherAI/gpt-j-6B', 'device': 'cuda:0', 'num_return_sequences': 1, 'batch_size': 1, 'length_penalty': 0, 'doc_stride': 0, 'factor_multiply_top_k': 0, 'handle_impossible_answer': False, 'max_answer_length': 200, 'max_seq_len': 2048}


In [10]:
response = reader.answer_one_question(texto_pergunta, texto_contexto)

The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Setting `pad_token_id` to `eos_token_id`:50256 for open-end generation.


chars(prompt): 487
tokens(prompt): 181
texto gerado: [{'generated_text': ' Acho que são os do Flamengo\n'}]


In [13]:
parm_dict_config['num_top_k'] = 4
reader.atualiza_parametros_resposta(parm_dict_config)

In [14]:
response = reader.answer_one_question(texto_pergunta, texto_contexto)

The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Setting `pad_token_id` to `eos_token_id`:50256 for open-end generation.


chars(prompt): 487
tokens(prompt): 181
texto gerado: [{'generated_text': ' O Flamengo, sem mais delongas.'}, {'generated_text': ' Flamengo\n\nPreencervaça a tab'}, {'generated_text': ' \nQualidade - Flamengo\nQuantid'}, {'generated_text': ' Por causa da chuva, o Flamengo'}]


In [16]:
parm_dict_config['num_top_k'] = 3
reader.atualiza_parametros_resposta(parm_dict_config)
response = reader.answer_one_question(texto_pergunta, texto_contexto)

The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Setting `pad_token_id` to `eos_token_id`:50256 for open-end generation.


chars(prompt): 494
tokens(prompt): 184
texto gerado: [{'generated_text': ' O time do Flamengo.'}, {'generated_text': ' Se você, algum'}, {'generated_text': ' Porca porra dos caras'}]


In [20]:
parm_dict_config['generator_temperature'] = 0.1
reader.atualiza_parametros_resposta(parm_dict_config)
response = reader.answer_one_question(texto_pergunta, texto_contexto)

The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Setting `pad_token_id` to `eos_token_id`:50256 for open-end generation.


chars(prompt): 494
tokens(prompt): 184
texto gerado: [{'generated_text': ' Flamengo\n'}, {'generated_text': ' O Flamengo'}, {'generated_text': ' Flamengo\n'}]


In [21]:
parm_dict_config['generator_sample'] = False
reader.atualiza_parametros_resposta(parm_dict_config)
response = reader.answer_one_question(texto_pergunta, texto_contexto)

The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Setting `pad_token_id` to `eos_token_id`:50256 for open-end generation.


chars(prompt): 494
tokens(prompt): 184


ValueError: num_return_sequences has to be 1, but is 3 when doing greedy search.

In [22]:
parm_dict_config['num_top_k'] = 1
reader.atualiza_parametros_resposta(parm_dict_config)
response = reader.answer_one_question(texto_pergunta, texto_contexto)

The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Setting `pad_token_id` to `eos_token_id`:50256 for open-end generation.


chars(prompt): 494
tokens(prompt): 184
texto gerado: [{'generated_text': ' O Flamengo.'}]


In [ ]:
import importlib
importlib.reload(trata_reader_pipie)

EleutherAI/gpt-j-6B

In [ ]:
teste_trata_reader_pipie.teste_context_reader(name_model = "EleutherAI/gpt-j-6B")

gpt-neo-2.7B

len(prompt): 487

pipe (trata_reader_pipie)
carga do modelo : 27s
gpu: 12575mb



answer_max_length: 100
    Tempo: menos de 1s
    texto gerado 1 letra

answer_max_length: 200
    texto gerado 5 letras

answer_max_length: 2048

Geração de texto:
    Tempo: 100 s
    Gpu: chegou a 23gb

    Resposta enormeeeee


In [ ]:
teste_trata_reader_pipie.teste_context_reader(name_model = 'EleutherAI/gpt-neo-2.7B')

gpt-neo-1.3b

In [ ]:
teste_trata_reader_pipie.teste_context_reader()


In [ ]:
squad_related.avaliar_dataset_squad_1_1(squad_dataset_pt)

In [ ]:
squad_related.avaliar_dataset_squad_1_1(squad_dataset_en)

## Avaliação

In [ ]:
import importlib
importlib.reload(squad_evaluate_v1_1)

In [ ]:
dict_config = {"doc_stride":128,\
               "top_k":3, \
               "max_answer_length":30, \
               "handle_impossible_answer":False}

In [ ]:
resultado_en = squad_evaluate_v1_1.evaluate_transfer(parm_dataset=squad_dataset_en, parm_dict_config=dict_config)
print(resultado_en)

In [ ]:
raise Exception(f"Cells bellow can not be executed. It is just for history reasons.")

## Outas execuções

In [ ]:
resultado_pt = squad_evaluate_v1_1.evaluate_transfer(squad_dataset_pt)
print(resultado_pt)

In [ ]:
resultado_en = squad_evaluate_v1_1.evaluate_transfer(squad_dataset_en)
print(resultado_en)

 self.doc_stride = 30
 
 tamanho_max_resposta = 40
 
 self.handle_impossible_answer = False

In [ ]:
resultado_en = squad_evaluate_v1_1.evaluate_transfer(squad_dataset_en)
print(resultado_en)

In [ ]:
resultado_pt = squad_evaluate_v1_1.evaluate_transfer(squad_dataset_pt)
print(resultado_pt)